File: mydriver-backend/src/models/booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({
  customerId: { type: String, required: true },
  driverId: { type: String },
  source: { type: String, required: true },
  destination: { type: String, required: true },
  vehicleType: { type: String, required: true },
  status: { type: String, enum: ['pending', 'accepted', 'completed', 'cancelled'], default: 'pending' },
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model('Booking', bookingSchema);

File: mydriver/lib/core/models/booking.dart
class Booking {
  final String id;
  final String customerId;
  final String? driverId;
  final String source;
  final String destination;
  final String vehicleType;
  String status;

  Booking({
    required this.id,
    required this.customerId,
    this.driverId,
    required this.source,
    required this.destination,
    required this.vehicleType,
    required this.status,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['_id'],
      customerId: json['customerId'],
      driverId: json['driverId'],
      source: json['source'],
      destination: json['destination'],
      vehicleType: json['vehicleType'],
      status: json['status'],
    );
  }
}